recipes/freertos-kernel/all/conanfile.py

```diff
@@ -0,0 +1,403 @@
+import os
+import textwrap
+
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.cmake import cmake_layout, CMake, CMakeToolchain
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, save
+
+required_conan_version = ">=1.60.0"
+
+
+
+default_configs = {
+    "GCC_POSIX": textwrap.dedent("""\
+        #ifndef FREERTOS_CONFIG_H
```

Comment:
I came across an interesting article regarding target_sources
target_sources()

https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/

If i saw it right the whole FreeRTOS is already and interface library. Is it possible to add the FreeRTOSconfig.h via target_sources within the application itself?
Would it be possible to add something like this to the actual application?

```
find_library(FREERTOS_LIB FreeRTOS::FreeRTOS)
target_sources(${FREERTOS_LIB} "FreeRTOSConfig.h")
```

Reply:
I don't think that work when the recipe itself builds FreeRTOS because `FreeRTOSConfig.h` needs to be made available to the recipe when it is being built.

The difficulties of handling `FreeRTOSConfig.h` directly are determining how to pass `FreeRTOSConfig.h` to Conan such that it is exported as part of the recipe, its contents are included as part of the package_id to distinguish the binary, and used to compile the recipe itself.

Reply:
on second thought you are probably right :(